src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -2,96 +2,143 @@
+}
+
+void BallPlacementPlayFSM::setPickOffDest(const BallPlacementPlayFSM::Update &event)
+{
+    Point ball_pos           = event.common.world_ptr->ball().position();
+    Rectangle field_boundary = event.common.world_ptr->field().fieldBoundary();
+
+    std::pair<Angle, Point> location = calculateWallPickOffDest(ball_pos, field_boundary);
+
+    pickoff_final_orientation = location.first;
+    pickoff_destination =
+        location.second - Vector::createFromAngle(pickoff_final_orientation)
+                              .normalize(BACK_AWAY_FROM_WALL_M);
+}
+
+void BallPlacementPlayFSM::pickOffWall(const BallPlacementPlayFSM::Update &event)
+{
+    PriorityTacticVector tactics_to_run = {{pickoff_wall_tactic}};
+
+    pickoff_wall_tactic->updateControlParams(
+        pickoff_destination, pickoff_final_orientation, true,
+        TbotsProto::MaxAllowedSpeedMode::BALL_PLACEMENT_WALL_DRIBBLE,
+        TbotsProto::MaxAllowedSpeedMode::BALL_PLACEMENT_WALL_DRIBBLE);
 
-    Angle kick_angle = calculateWallKickoffAngle(ball_pos, field_lines);
-    pivot_kick_tactic->updateControlParams(ball_pos, kick_angle, auto_chick);
-    tactics_to_run[0].emplace_back(pivot_kick_tactic);
+    setupMoveTactics(event, event.common.num_tactics -
+                                static_cast<unsigned int>(tactics_to_run[0].size()));
+    tactics_to_run[0].insert(tactics_to_run[0].end(), move_tactics.begin(),
+                             move_tactics.end());
 
     event.common.set_tactics(tactics_to_run);
 }
 
 void BallPlacementPlayFSM::alignPlacement(const Update &event)
 {
+    PriorityTacticVector tactics_to_run = {{}};
+
     std::optional<Point> placement_point =
         event.common.world_ptr->gameState().getBallPlacementPoint();
 
     if (placement_point.has_value())
     {
-        PriorityTacticVector tactics_to_run = {{}};
-
-        // setup move tactics for robots away from ball placing robot
-        setupMoveTactics(event);
-        tactics_to_run[0].insert(tactics_to_run[0].end(), move_tactics.begin(),
-                                 move_tactics.end());
-
         // find position behind the ball where the ball is aligned directly in front
         // placement point from the placing robot's POV
         Vector alignment_vector =
             (placement_point.value() - event.common.world_ptr->ball().position())
                 .normalize();
         Angle setup_angle = alignment_vector.orientation();
         setup_point       = event.common.world_ptr->ball().position() -
-                      2 * alignment_vector * ROBOT_MAX_RADIUS_METERS;
+                      ALIGNMENT_VECTOR_LENGTH_FACTOR * alignment_vector;
 
         align_placement_tactic->updateControlParams(
             setup_point, setup_angle, TbotsProto::DribblerMode::OFF,
             TbotsProto::BallCollisionType::AVOID, {AutoChipOrKickMode::OFF, 0},
-            TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+            TbotsProto::MaxAllowedSpeedMode::STOP_COMMAND,
```

Comment:
instead of re-using `STOP_COMMAND`, then we should create a new `DRIBBLE` or `SLOW` speed mode.